src/nvim/testdir/test_autocmd.vim

```diff
@@ -786,7 +786,6 @@ endfunc
 " Test for autocommand that deletes the current buffer on BufLeave event.
 " Also test deleting the last buffer, should give a new, empty buffer.
 func Test_BufLeave_Wipe()
-  throw 'skipped: TODO: '
```

Comment:
```

[OLDTEST] Running test_autoload
=================================================================
==22389==ERROR: AddressSanitizer: heap-use-after-free on address 0x6260001891c8 at pc 0x000000b41bea bp 0x7ffc3cc181d0 sp 0x7ffc3cc181c8
READ of size 4 at 0x6260001891c8 thread T0
    #0 0xb41be9 in _typval_encode_nothing_convert_one_value /home/travis/build/neovim/neovim/build/../src/nvim/eval/typval_encode.c.h:320:15
    #1 0xb3cd6f in encode_vim_to_nothing /home/travis/build/neovim/neovim/build/../src/nvim/eval/typval_encode.c.h:833:9
    #2 0xb0dbec in tv_clear /home/travis/build/neovim/neovim/build/../src/nvim/eval/typval.c:2201:25
    #3 0x93c2e8 in vars_clear_ext /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:19381:9
    #4 0x8c8d39 in vars_clear /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:19358:3
    #5 0x8c87c7 in eval_clear /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:640:3
    #6 0x1101e6f in free_all_mem /home/travis/build/neovim/neovim/build/../src/nvim/memory.c:669:3
    #7 0x143e32c in mch_exit /home/travis/build/neovim/neovim/build/../src/nvim/os_unix.c:155:3
    #8 0xfe1bb4 in getout /home/travis/build/neovim/neovim/build/../src/nvim/main.c:702:3
    #9 0xce559e in ex_quit_all /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:6141:5
    #10 0xc6d2f0 in do_one_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:2242:5
    #11 0xc4ba87 in do_cmdline /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:593:20
    #12 0x9225ec in call_user_func /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:21837:3
    #13 0x8ec29b in call_func /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6392:11
    #14 0x901a13 in get_func_tv /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:6149:11
    #15 0x8fa59e in ex_call /home/travis/build/neovim/neovim/build/../src/nvim/eval.c:2744:9
    #16 0xc6d2f0 in do_one_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:2242:5
    #17 0xc4ba87 in do_cmdline /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:593:20
    #18 0xc318d0 in do_source /home/travis/build/neovim/neovim/build/../src/nvim/ex_cmds2.c:3228:3
    #19 0xc2d704 in cmd_source /home/travis/build/neovim/neovim/build/../src/nvim/ex_cmds2.c:2973:14
    #20 0xc2dec0 in ex_source /home/travis/build/neovim/neovim/build/../src/nvim/ex_cmds2.c:2954:3
    #21 0xc6d2f0 in do_one_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:2242:5
    #22 0xc4ba87 in do_cmdline /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:593:20
    #23 0xc51c45 in do_cmdline_cmd /home/travis/build/neovim/neovim/build/../src/nvim/ex_docmd.c:269:10
    #24 0xfdef3b in exe_commands /home/travis/build/neovim/neovim/build/../src/nvim/main.c:1706:5
    #25 0xfcc2a1 in main /home/travis/build/neovim/neovim/build/../src/nvim/main.c:548:5
    #26 0x7fc9621ac82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #27 0x4507e8 in _start (/home/travis/build/neovim/neovim/build/bin/nvim+0x4507e8)
```

Reply:
see 66f5e5c7d7ce762db71886b72e759a745c2b41ce for explanation (and its parent, 9af9ea6099d109982d642885c6b7fc5e5082c3f8 for a similar case). The bug was already present, but I didn't want it to block merging of many related patches.  

Maybe when we  are fully caught up to 8.0.1223 the bug will be fixed. Else we will need to look more closely.